src/components/CourseModal/CourseModal.tsx

```diff
@@ -16,29 +17,39 @@ import { DictProps } from '@/lib/i18n';
 import { useTranslation } from '@/lib/i18n/client';
 import { useSession } from 'next-auth/react';
 import Loading from '@/app/[lang]/loading';
+import { UserNamesAndIds } from '@/lib/prisma/users';
 import { usePathname, useRouter, useSearchParams } from 'next/navigation';
+import { useState, useEffect } from 'react';
+import TrainerTools from './TrainerTools';
+import { isTrainerOrAdmin } from '@/lib/auth-utils';
 
 interface Props extends DictProps {
   course: CourseWithTagsAndStudentCount | undefined;
   usersEnrolledCourseIds: string[];
+  enrolledStudents: UserNamesAndIds | null;
   enrolls: string;
-  description: string;
   editCourseLabel: string;
 }
 
 export default function CourseModal({
   course,
   usersEnrolledCourseIds,
+  enrolledStudents,
   lang,
   enrolls,
-  description,
   editCourseLabel,
 }: Props) {
   const { t } = useTranslation(lang, 'components');
   const { data: session, status } = useSession({ required: true });
   const router = useRouter();
   const searchParams = useSearchParams();
   const pathname = usePathname();
+  const [courseView, setCourseView] = useState<string | null>('details');
```

Comment:
Typescript can infer the type with this one.

```suggestion
  const [courseView, setCourseView] = useState('details');
```